File: sdk/nodejs/src/resources/domains.js
"use strict";

const { BaseResource } = require("./base");

class DomainsResource extends BaseResource {
  async list(params = {}) {
    return this._http.get("/api/domain", params);
  }

  async get(domainId) {
    return this._http.get(`/api/domain/${domainId}`);
  }

  async getByName(name) {
    return this._http.get(`/api/domain/name/${name}`);
  }

  async whois(domain) {
    return this._http.get(`/api/whois/${domain}`);
  }

  async whoisLookup(domain) {
    return this._http.get(`/api/whoislookup/${domain}`);
  }

  async checkAvailability(domain) {
    return this._http.get(`/api/domain/availability/${domain}`);
  }

  async getNameservers(domainId) {
    return this._http.get(`/api/domain/${domainId}/ns`);
  }

  async updateNameservers(domainId, ns) {
    return this._http.put(`/api/domain/${domainId}/ns`, ns);
  }

  async getEppCode(domainId) {
    return this._http.get(`/api/domain/${domainId}/epp`);
  }

  async sync(domainId) {
    return this._http.get(`/api/domain/${domainId}/sync`);
  }

  async getLock(domainId) {
    return this._http.get(`/api/domain/${domainId}/reglock`);
  }

  async updateLock(domainId, reglock) {
    return this._http.put(`/api/domain/${domainId}/reglock`, { reglock });
  }

  async updateIdProtection(domainId, idprotection) {
    return this._http.put(`/api/domain/${domainId}/idprotection`, { idprotection });
  }

  async getContact(domainId) {
    return this._http.get(`/api/domain/${domainId}/contact`);
  }

  async updateContact(domainId, data) {
    return this._http.put(`/api/domain/${domainId}/contact`, data);
  }

  async getEmailForwarding(domainId) {
    return this._http.get(`/api/domain/${domainId}/email/forwarding`);
  }

  async updateEmailForwarding(domainId, entries) {
    return this._http.put(`/api/domain/${domainId}/email/forwarding`, { entries });
  }

  async getAvailableTlds() {
    return this._http.get("/api/domain/order");
  }

  async order(domain, years = 1, data = {}) {
    return this._http.post("/api/domain/order", { domain, years, ...data });
  }

  async renew(domainId, years = 1) {
    return this._http.post(`/api/domain/${domainId}/renew`, { years });
  }

  async getTldForm(tldId) {
    return this._http.get(`/api/domain/order/${tldId}/form`);
  }

  async getDocuments(domainId) {
    return this._http.get(`/api/domain/${domainId}/documents`);
  }
}

module.exports = { DomainsResource };

File: sdk/csharp/Resources/Domains.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HiTechCloud.SDK.Resources;

public class Domains : BaseResource
{
    public Domains(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> WhoisAsync(string domain)
    {
        return await Http.GetAsync(string.Format("/api/whois/{domain}", domain));
    }

    public async Task<Dictionary<string, object?>> DomainsAsync()
    {
        return await Http.GetAsync("/api/domain");
    }

    public async Task<Dictionary<string, object?>> DomainAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/domain/{id}", id));
    }

    public async Task<Dictionary<string, object?>> DomainByNameAsync(string name)
    {
        return await Http.GetAsync(string.Format("/api/domain/name/{name}", name));
    }

    public async Task<Dictionary<string, object?>> NameserversAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/domain/{id}/ns", id));
    }

    public async Task<Dictionary<string, object?>> UpdateNameserversAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PutAsync(string.Format("/api/domain/{id}/ns", id), data);
    }

    public async Task<Dictionary<string, object?>> EppCodeAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/domain/{id}/epp", id));
    }

    public async Task<Dictionary<string, object?>> SyncDomainAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/domain/{id}/sync", id));
    }

    public async Task<Dictionary<string, object?>> ReglockAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/domain/{id}/reglock", id));
    }

    public async Task<Dictionary<string, object?>> UpdateReglockAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PutAsync(string.Format("/api/domain/{id}/reglock", id), data);
    }

    public async Task<Dictionary<string, object?>> IdProtectionAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PutAsync(string.Format("/api/domain/{id}/idprotection", id), data);
    }

    public async Task<Dictionary<string, object?>> DomainContactsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/domain/{id}/contact", id));
    }

    public async Task<Dictionary<string, object?>> UpdateContactsAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PutAsync(string.Format("/api/domain/{id}/contact", id), data);
    }

    public async Task<Dictionary<string, object?>> EmailForwardingAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/domain/{id}/email/forwarding", id));
    }

    public async Task<Dictionary<string, object?>> UpdateEmailForwardingAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PutAsync(string.Format("/api/domain/{id}/email/forwarding", id), data);
    }

    public async Task<Dictionary<string, object?>> AvailabilityAsync(string domain)
    {
        return await Http.GetAsync(string.Format("/api/domain/availability/{domain}", domain));
    }

    public async Task<Dictionary<string, object?>> OrdersAsync()
    {
        return await Http.GetAsync("/api/domain/order");
    }

    public async Task<Dictionary<string, object?>> CreateOrderAsync(Dictionary<string, object> data)
    {
        return await Http.PostAsync("/api/domain/order", data);
    }

    public async Task<Dictionary<string, object?>> OrderFormAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/domain/order/{id}/form", id));
    }

    public async Task<Dictionary<string, object?>> RenewDomainAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/domain/{id}/renew", id), data);
    }

    public async Task<Dictionary<string, object?>> WhoisLookupAsync(string domain)
    {
        return await Http.GetAsync(string.Format("/api/whoislookup/{domain}", domain));
    }

    public async Task<Dictionary<string, object?>> DocumentsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/domain/{id}/documents", id));
    }
}
